com.unity.template.renderstreaming-hd/Assets/Scripts/Signaling.cs

```diff
@@ -1,272 +1,543 @@
+
+
+        public event OnSignedInHandler OnSignedIn;
+        public event OnOfferHandler OnOffer;
+        public event OnIceCandidateHandler OnIceCandidate;
+
+        public Signaling(string url, float timeout) {
+            _uri = new Uri(url);
+            _timeout = timeout;
+            _wsCloseEvent = new AutoResetEvent(false);
+        }
+
+        //------------------------------------------------------------------------------------------
+        //Global stuff
+        //------------------------------------------------------------------------------------------
+
+
+        public void Start() {
+
+            _running = true;
+            if (_uri.Scheme == "http" || _uri.Scheme == "https") {
+                _signalingThread = new Thread(HTTPPooling);
+                _signalingThread.Start();
+            } else {
+                _signalingThread = new Thread(WSManage);
+                _signalingThread.Start();
+            }
+        }
+
+
+        public void Stop() {
+
+            _running = false;
+
+            if (_uri.Scheme == "ws" || _uri.Scheme == "wss") {
+                if (_webSocket != null) {
+                    _webSocket.Close();
+                }
+            }
+        }
+
+        public void SendCandidate(string connectionId, RTCIceCandidate candidate) {
+
+            CandidateData data = new CandidateData();
+            data.connectionId = connectionId;
+            data.candidate = candidate.candidate;
+            data.sdpMLineIndex = candidate.sdpMLineIndex;
+            data.sdpMid = candidate.sdpMid;
+
+            if (_uri.Scheme == "http" || _uri.Scheme == "https") {
+
+                HTTPPost("signaling/candidate", data);
+
+            } else {
+
+                RoutedMessage<CandidateData> routedMessage = new RoutedMessage<CandidateData>();
+                routedMessage.to = connectionId;
+                routedMessage.message = data;
+
+                WSSend(routedMessage);
+            }
+
+        }
+
+        public void SendAnswer(string connectionId, RTCSessionDescription answer) {
+
+            DescData data = new DescData();
+            data.connectionId = connectionId;
+            data.sdp = answer.sdp;
+            data.type = "answer";
+
+            if (_uri.Scheme == "http" || _uri.Scheme == "https") {
+                HTTPPost("signaling/answer", data);
+            } else {
+
+                RoutedMessage<DescData> routedMessage = new RoutedMessage<DescData>();
+                routedMessage.to = connectionId;
+                routedMessage.message = data;
+
+                WSSend(routedMessage);
+            }
+
+        }
+
+
+        //------------------------------------------------------------------------------------------
+        //WebSocket Version
+        //------------------------------------------------------------------------------------------
+
+
+        public void WSManage() {
```

Comment:
Is it possible to put the HTTP implementation and WS implementation to separate files, instead of combining them in this one class ?
So like maybe:
* Signaling: the "manager"
* WSImpl: the websocket implementation
* HTTPImpl: the HTTP Implementation

Reply:
I separate implementation this PR https://github.com/Unity-Technologies/UnityRenderStreaming/pull/253 , is it okey?

Reply:
OK